src/llmcompressor/modifiers/utils/hooks.py

```diff
@@ -23,24 +23,42 @@ class HooksMixin(BaseModel):
 
     Hooks can be applied to modules or parameters
 
-    Lifecycle:
-        - modifier.register_forward_hook(module, hook)
-        - with HooksMixin.disable_hooks(): model.forward()
-        - modifier.remove_hooks()
+    Typical example
+    >>> modifier.register_forward_hook(module, hook)
+    >>> with HooksMixin.disable_hooks():
+            model.forward(...)
+    >>> modifier.remove_hooks()
+
+    Example of activating only a specific subset of hooks
+    >>> hooks = [modifier.register_forward_hook(module, hook) for module in ...]
+    >>> with HooksMixin.disable_hooks(keep=hooks):
+            model.forward(...)
+    >>> modifier.remove_hooks(hooks)
     """
 
-    _HOOKS_DISABLED: ClassVar[bool] = False  # attached to global HooksMixin
-    _hooks: Set[RemovableHandle] = set()  # attached to local subclasses
+    # attached to global HooksMixin class
+    _HOOKS_DISABLED: ClassVar[bool] = False
+    _HOOKS_KEEP_ENABLED: ClassVar[Set[RemovableHandle]] = set()
+
+    # attached to local subclasses
+    _hooks: Set[RemovableHandle] = set()
 
     @classmethod
     @contextlib.contextmanager
-    def disable_hooks(cls):
-        """Disable all hooks across all modifiers"""
+    def disable_hooks(cls, keep: Set[RemovableHandle] = set()):
```

Comment:
you're not mutating `keep` below, so this won't be an issue here, but python has very weird behavior when you use default container values as inputs to functions:

```python
def f(m = []):
    m.append(4)
    return m

print(f()) #prints "[4]"
print(f()) #prints "[4,4]"!
```

might be better to have `keep: Set[RemoveableHandle] | None = None` instead, and just do the `if keep is None` check.

Reply:
Totally agree. In the future we should add this to the linter options, no mutable defaults